cogs/games.py

```diff
@@ -108,13 +128,50 @@ def __init__(self, bot):
 		self.config = ConfigTable(bot, 'trivia', ('guild_id', 'user_id'))
 		self.playing = set()
 
+		asyncio.run_coroutine_threadsafe(self.get_trivia_categories(), self.bot.loop)
+
+	async def get_trivia_categories(self):
+		global TRIVIA_CATEGORIES
+
+		try:
+			async with self.bot.aiohttp.get(API_CATEGORY_LIST_URL) as resp:
+				if resp.status != 200:
+					return
+		
+				res = await resp.json()
+		except asyncio.TimeoutError:
+			return
+
+		categories_id_map = {}
+
+		for category in res['trivia_categories']:
+			name = category['name'].lower() # Name is sometimes something like `GeneralCategory: SpecificCategory`, like `Entertainment: Books`, so we just trim that out if it's there
+			colon_pos = name.find(':')
+			trimmed_name = name[0 if colon_pos == -1 else colon_pos + 2:] # Cut out both the colon, and the space after it
+
+			if trimmed_name != name:
+				categories_id_map[name] = category['id'] # However, if it is a XXXX: YYY category, we do include both names
+
+			categories_id_map[trimmed_name] = category['id'] # Map the name to ID, so a user can enter a name instead of an ID
+
+		categories_id_map['anime'] = categories_id_map['japanese anime & manga'] # Some manual entries, since the API picks dumb long names for everything
+		categories_id_map['science'] = categories_id_map['science & nature']
+
+		TRIVIA_CATEGORIES = categories_id_map
+
 	@commands.group(invoke_without_command=True, cooldown_after_parsing=True)
 	@commands.bot_has_permissions(embed_links=True, add_reactions=True)
 	@commands.cooldown(rate=1, per=300.0, type=commands.BucketType.member)
-	async def trivia(self, ctx, *, difficulty: DifficultyConverter = None):
-		'''Trivia time! Optionally specify a difficulty as argument. Valid difficulties are `easy`, `medium` and `hard`.'''
+	async def trivia(self, ctx, category: CategoryConverter = None, *, difficulty: DifficultyConverter = None):
```

Comment:
I didn't use a union, since I noticed that `typing` also provides `Optional`, which I think fits much better since the category *is* an optional parameter, and should be treated as difficulty when the converter fails (which is pretty much exactly what I manually implemented with all the ugly converter logic)